### ISSUE TITLE
Parameterized declarations mess up syntax highlighting

### ISSUE THREAD

**Issue by djoffe** [eirikpre/VSCode-SystemVerilog#23]
In the following code example, lines 2 and 3 (the uvm_analysis_export declarations) mess up the lines following them. Comment those two lines and the rest of the code looks fine.
- Uncomment one line and the syntax highlighting is messed up until the first `new()` keyword following it (`function new(...)`).
- Uncomment the second line and the syntax highlighting is messed up until the second `new` keyword following it (`super.new(...)`).
- And so on for each parameterized declaration (The `#` seems to be causing the issue here)

```sv
class env extends uvm_env;
    uvm_analysis_export#(some_pkg::item)    some_export;
    uvm_analysis_export#(some_pkg::item)    some_other_export;

    // References
    reg_bus_uvc_pkg::env   reg_bus_env;
    reg_bus_uvc_pkg::agent reg_bus_agt;

    function new(string name, uvm_component parent);
        super.new(name, parent);

        some_export         = new("some_export", this);
        some_other_export   = new("some_other_export", this);
    endfunction
endclass
```

![image](https://user-images.githubusercontent.com/2979478/44029180-49bba104-9ef4-11e8-931e-ca220fa80971.png)
![image](https://user-images.githubusercontent.com/2979478/44029201-593c583a-9ef4-11e8-93ca-b1f985cd1807.png)
![image](https://user-images.githubusercontent.com/2979478/44029223-670ed3fc-9ef4-11e8-8d09-ed8f62242c6f.png)

**Comment by djoffe** [comment]
Correction, it is not the `new` keyword that closes the bad highlighting area, but any opening paren `(`.

**Comment by eirikpre** [comment]
Yes, I know this is an issue. Also when you have extra paranthesis in a portlist it doesn't match the number of parantheses.
It has been a while since I last tinkered with the textmate setup. I was kinda hoping we could have only one source for the syntax highlighting.
I'll fix this the next time I'm working with the extension :)

**Comment by eirikpre** [comment]
Fixed with 0.4.1 (released now, thanks for reporting)
